java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBFileReferenceStore.java

```diff
@@ -201,6 +201,9 @@ private List<TransactWriteItem> splitFileReferenceWrites(SplitFileReferenceReque
     @Override
     public void atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFiles(
             String jobId, String partitionId, List<String> filesToBeMarkedReadyForGC, List<FileReference> newFiles) throws StateStoreException {
+        if (newFiles.stream().map(FileReference::getFilename).anyMatch(filesToBeMarkedReadyForGC::contains)) {
+            throw new StateStoreException("One or more new files has same filename as file to be marked as ready for GC");
+        }
```

Comment:
I think this check means that the logic below for resolving update references is no longer needed, with the variable updateReferencesByFilename. We can just add a write item to decrement the references for each reference being deleted, and one to increment the references for each file being added.

Reply:
Thanks, i've removed the map and update the reference count for each file now